test/allEqual.js

```diff
@@ -0,0 +1,29 @@
+import { assert } from 'chai';
+
+import * as RA from '../src/index';
+
+describe('allEqual', function() {
+  context('when all items are equal', function() {
+    specify('returns true', function() {
+      assert.isTrue(RA.allEqual([4, 4, 4, 4]));
+    });
+  });
+
+  context('when all items are deeply equal', function() {
+    specify('should return true', function() {
+      assert.isTrue(RA.allEqual([{ key: 'foo' }, { key: 'foo' }]));
+    });
+  });
+
+  context('when items are not equal', function() {
+    specify('should return false', function() {
+      assert.isFalse(RA.allEqual([1, 1, 2, 1, 1]));
+    });
+  });
+
+  context('when empty list provided', function() {
+    specify('should return true', function() {
+      assert.isTrue(RA.allEqual([]));
+    });
```

Comment:
What happens if no value is passed: `allEqual()`?
What happens if the same function is passed multiple times?
```
const f = () => {}
RA.allEqual([f, f, f]))
```

Reply:
Can add test for testing functions. 

> What happens if no value is passed: allEqual()?
I think we don't care. The signature is clear and everything outside of the signature
is unspecified.